packages/flutter/lib/src/rendering/sliver_multi_box_adaptor.dart

```diff
@@ -241,6 +241,40 @@ abstract class RenderSliverMultiBoxAdaptor extends RenderSliver
     }());
   }
 
+  @override
+  void insert(RenderBox child, { RenderBox after }) {
+    assert(!_keepAliveBucket.containsValue(child));
+    super.insert(child, after: after);
+    assert(firstChild != null);
+    if (integrityEnabled){
+      verifyIntegrity();
+    }
+  }
+
+  /// Move child to a different location in the list.
+  /// There are two scenarios:
+  /// 1, the child is in _keepAliveBucket
+  /// In this case, the child is no longer in the rendering tree. we just
```

Comment:
I think I was confused by this, you're right, the chid is no longer in the child list maintained by the `ContainerRenderObjectMixin`, but it is still considered a child by the `AbstractNode` superclass. Maybe call out the specific child list you mean to avoid confusion?

Reply:
fixed